src/humidity.jl

```diff
@@ -0,0 +1,63 @@
+"""
+Compute the saturation specific humidity for a single atmospheric level.
+"""
+function get_saturation_specific_humidity(
+    T::Array{NF,2},  # Absolute temperature for a single atmospheric level
+    p::Array{NF,2},  # Normalised surface pressure
+    σ::NF,           # σ of the current atmospheric level
+    M::ModelSetup{NF},
+) where {NF<:AbstractFloat}
+    @unpack nlon, nlat = M.geospectral.geometry
+    @unpack e₀, C₁, C₂, T₀, T₁, T₂ = M.params.humidity_coefs
+
+    Qsat = zeros(nlon, nlat)  # Saturation specific humidity
+
+    @inbounds for j = 1:nlat, i = 1:nlon
+        if T[i, j] > T₀
+            Qsat[i, j] = e₀ * exp(C₁ * (T[i, j] - T₀) / (T[i, j] - T₁))
+        else
+            Qsat[i, j] = e₀ * exp(C₂ * (T[i, j] - T₀) / (T[i, j] - T₂))
+        end
+    end
+
+    if σ <= 0
+        @. Qsat = 622.0 * Qsat / (p[1, 1] - 0.378 * Qsat)
+    else
+        @. Qsat = 622.0 * Qsat / (σ * p - 0.378 * Qsat)
+    end
+
+    return Qsat
+end
+
+"""
+Compute the saturation specific humidity.
+"""
+function get_saturation_specific_humidity(
+    T::Array{NF,3},  # Absolute temperature
```

Comment:
My idea was to move every vertical loop to `src/distributed_vertical.jl` such that we have one place where in the end the `@distributed` will distribute the vertical layers across processors for parallelisation. However, the question whether we actually want to do that or whether the parametrizations should be distributed in the horizontal. In the latter I'd suggest to create a `src/distributed_horizontal.jl` as a central place for the horizontal domain decomposition. Tricky!